modules/nextflow/src/main/groovy/nextflow/extension/PublishOp.groovy

```diff
@@ -189,42 +190,55 @@ class PublishOp {
     }
 
     /**
-     * Once all values have been published, publish the index
-     * and write it to a file (if enabled).
+     * Get or resolve the annotations of a workflow output.
+     *
+     * @param annotations Map | Closure<Map>
+     * @param value
+     */
+    protected static Map getAnnotations(annotations, value) {
```

Comment:
Still not fully sure about this. Wonder if it should not be a list of string and call it `labels` for consistency with process labels. this would also make it possible to propagate the process labels to the TaskRun model

Reply:
You should explore that in a separate PR. This change is just fixing a bug with the current behavior

Reply:
I think for the new model (with lists instead of maps), the list of strings will be easier to refer in a query, now we have to specify `annotations.key` and `annotations.value`. I can do it in a separate PR once it is merged.

Reply:
> You should explore that in a separate PR

Agree  

> I think for the new model (with lists instead of maps), the list of strings will be easier to refer in a query

let's do

Reply:
@pditommaso I assume that you also want to change `annotations` to `labels` in the Lineage model, right? 
In processes there are two directives `label` and `resourceLabels`. The one that you want to add in TaskRun model is the first one, isn't it? Or do you refer to another property?

Reply:
Why would we save the process labels in the lineage model? They are used only to apply configuration like cpus, memory, gpus, etc. They are mostly execution details

Reply:
Because it's metadata and to avoid introducing yer another notation other `label` and `resourceLabels`

Reply:
There is also `tag` that exist in process and publishDir. I think it is the best one to be reused. I remember we had this discussion and we decided to keep `annotations` for M1

Reply:
moreover  .. tho tags has always been kind of experimental

Reply:
What if the user doesn't want the process labels to be included? They might want to use these annotations for a different purpose and forcing them to include the process labels could muddle that purpose